arithmetization/src/main/java/net/consensys/linea/zktracer/module/bls/BlsOperation.java

```diff
@@ -0,0 +1,1094 @@
+      final Bytes aX0 = callData.slice(3 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aY3 = callData.slice(4 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aY2 = callData.slice(5 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aY1 = callData.slice(6 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aY0 = callData.slice(7 * LLARGE + sizeOffset, LLARGE);
+
+      final boolean wellFormedCoordinate =
+          wellFormedFpCoordinate(indexOffset, aX3, aX2, aX1, aX0, aY3, aY2, aY1, aY0);
+      final boolean isSmallPointInSubgroup =
+          isSmallPointInSubGroup(indexOffset, aX3, aX2, aX1, aX0, aY3, aY2, aY1, aY0);
+      final boolean mextBit = wellFormedCoordinate && !isSmallPointInSubgroup;
+      Preconditions.checkArgument(mextBit == (wellFormedCoordinate && !successBit));
+
+      if (mextBit && !mextBitIsSet) {
+        for (int j = 0; j <= CT_MAX_SMALL_POINT; j++) {
+          this.mextBit.set(indexOffset + j, true);
+        }
+        mextBitIsSet = true;
+      }
+    }
+  }
+
+  private void handleBlsG2Add() {
+    boolean mextBitIsSet = false;
+
+    for (int k = 0; k < 2; k++) {
+      final int sizeOffset = k * SIZE_LARGE_POINT;
+      final int indexOffset = k * (CT_MAX_LARGE_POINT + 1);
+
+      // Extract inputs
+      final Bytes aXIm3 = callData.slice(sizeOffset, LLARGE);
+      final Bytes aXIm2 = callData.slice(LLARGE + sizeOffset, LLARGE);
+      final Bytes aXIm1 = callData.slice(2 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aXIm0 = callData.slice(3 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aXRe3 = callData.slice(4 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aXRe2 = callData.slice(5 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aXRe1 = callData.slice(6 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aXRe0 = callData.slice(7 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYIm3 = callData.slice(8 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYIm2 = callData.slice(9 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYIm1 = callData.slice(10 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYIm0 = callData.slice(11 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYRe3 = callData.slice(12 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYRe2 = callData.slice(13 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYRe1 = callData.slice(14 * LLARGE + sizeOffset, LLARGE);
+      final Bytes aYRe0 = callData.slice(15 * LLARGE + sizeOffset, LLARGE);
+
+      final boolean wellFormedCoordinate =
+          wellFormedFp2Coordinate(
+              indexOffset,
+              aXIm3,
+              aXIm2,
+              aXIm1,
+              aXIm0,
+              aXRe3,
+              aXRe2,
+              aXRe1,
+              aXRe0,
+              aYIm3,
+              aYIm2,
+              aYIm1,
+              aYIm0,
+              aYRe3,
+              aYRe2,
+              aYRe1,
+              aYRe0);
```

Comment:
What I don't understand is why you cut up the coordinates at this stage. For instance
```java
              aXIm3,
              aXIm2,
              aXIm1,
              aXIm0,
```
Represents a 64 byte string which may or may not correspond to a single 48 byte field element. Why do the splitting here when
- the hi/lo stuff shouldn't matter to the implementation
- you end up recombining stuff e.g. in `wcpGeneralizedCallToLT`